### ISSUE TITLE
The use-case of `--wait` and having it enabled by default with `odo service create`

### ISSUE THREAD

**Issue by cdrage** [redhat-developer/odo#1144]
This is just to spur discussion, but whenever we push a component, by default, Odo will "wait" and the spinner will spin until it's completed.

This is not the case however with `odo service create`. Instead, we output:

```sh
▶ ./odo service create dh-postgresql-apb my-postgresql-db --plan dev -p postgresql_user=luke -p postgresql_password=secret

 ✓  Service 'my-postgresql-db' was created
Progress of the provisioning will not be reported and might take a long time.
You can see the current status by executing 'odo service list'
```

This is completely different than `odo create` or `odo push`.

I believe we should unify / have similar interactions with both commands. 

My suggestions:

 - By default, we should show the status of provisioning and have Odo wait until the service is deployed. Last time I deployed a database it took less than 30 seconds.
 - Show provisioning output by default when deploying (same as the above suggestion)
 - Have --wait an alias / the same as --show for the two commands, we should either have --wait OR --show for both commands, but not both.

ping @geoand @metacosm @kadel

**Comment by geoand** [comment]
Currently we have a wait flag on odo link (which is off by default).

I think the reason we didn't add wait to odo service create was that it
could take a long time without being an actual blocker in itself for the
user (since having a service on it's own probably won't be of much use to
odo users).
Link is the actual blocker and that's where I prefer we made odo wait by
default.

What do you all think?

**Comment by jorgemoralespou** [comment]
@cdrage I can show you many services that takes minutes to deploy :-( so I'm against having the CLI waiting for the service to be created. Think that it's way different deploying a service implemented as a template than as an APB. Also, think that in many cases, the real provisioning action happens asynchronously, so even if the broker/catalog tells you it's done, it's not, and that's dependant on the service implementation.

I'd rather have the `--wait` be an optional flag that the user passes whenever he knows is going to be fast. Then we have `odo service list` or we can implement `odo service status` to get information about a state of a specific service, but definitely not block the user for the service to be created.

I'm pretty sure that while the service is provisioned, the user will make use of the time for some other stuff (maybe creating a service or pushing some code).

> By default, we should show the status of provisioning and have Odo wait until the service is deployed. Last time I deployed a database it took less than 30 seconds.

I would definitely not do this.

>Show provisioning output by default when deploying (same as the above suggestion)

Have you looked at deploying services from the TSB as well as AB? As I said, this behave very differently, and the output you can get is quite different. What would you show here as output?

>Have --wait an alias / the same as --show for the two commands, we should either have --wait OR --show for both commands, but not both.

This last one I'm not following.

**Comment by hrishin** [comment]
> I'm pretty sure that while the service is provisioned, the user will make use of the time for some other stuff (maybe creating a service or pushing some code).


Quite a valid point @jorgemoralespou. However, it would be really nice (nice to have) if user could provide some option to `odo service <appropriate subcommand> --wait`. As a user, at some point in time surely I would require some way ot get service provisioning status instead of keep polling.

Or at least we can user doc with some tip like `watch odo service list`.

**Comment by jorgemoralespou** [comment]
I think we should probably introduce a command `odo service watch <servicename>` to wait for status.

**Comment by geoand** [comment]
In my PR I didn't add `odo service watch`. IMHO adding the `wait` flag to `odo service create` along with the existing `wait` flag in link should be more than enough :)
